lib/cocoapods/generate/podfile_generator.rb

```diff
@@ -189,6 +189,20 @@ def use_modular_headers?
         end
       end
 
+      # @return [Boolean, Hash]
+      #         the value to use for `use_frameworks!` DSL directive
+      #
+      def use_frameworks_value
+        return configuration.use_frameworks? unless configuration.use_podfile?
```

Comment:
should the configuration also have a `--use-static-frameworks` flag?

Reply:
it can if we want it to be a feature we want to add. I am mostly maintaining what was happening before and supporting CocoaPods 1.9 only here.

Reply:
I take it a bit back. This PR adds support for reading the `Podfile` value for `use_frameworks!` if the configuration asks to `use_podfile?`. It _also_ adds support for 1.9 with the custom linkage style.

Reply:
anything doable with a podfile should also be doable without a podfile

Reply:
This is not true for `inhibit_all_warnings` option.

Reply:
i.e these things should be somewhere declared/known if the project is supposed to work that way and not "random" PR commentary imo.

Reply:
will add it.